### ISSUE TITLE
Add TextDocument.buildTarget

### ISSUE THREAD

**Issue by xeno-by** [scalameta/scalameta#1307]
Looks like it would be beneficial to add something like `TextDocument.buildTarget` to keep track of build targets that generated the text document.

**Comment by olafurpg** [comment]
I'm a bit unsure about this one. One one side this could help distinguish between different snapshots (JS/JVM/2.11/2.12) of the same source file, but on the other side I feel like this can be handled outside of the semanticdb schema. This would have helped deduplicate js/jvm results in my community build corpus, and I guess you don't have to set if unless you want to.

**Comment by ShaneDelmore** [comment]
It could be useful to support multiple versions of a library, as well as to identify the source of the symbol to assist with imports or build file changes.

**Comment by xeno-by** [comment]
We also need this internally.

**Comment by olafurpg** [comment]
Fair enough. Is the `uri` field sufficient to encode this information? A URI could have query parameters `foo/Foo.scala?target=//path/foo:bar&hash=wjk23j4rhsaa`

**Comment by ShaneDelmore** [comment]
That is an interesting idea.  It leaves the data fairly unstructured but is a nice place to encode information that isn’t explicitly in the spec, although if we want it to be part of the spec we could call out exactly the query params to be used for this usage.  This gives the option to store information like compile parameters which could influence if a specific semanticdb entry is appropriate for answering a users question.  I will have to think on this a bit, but it sounds pretty reasonable right now.

**Comment by xeno-by** [comment]
Also the question is how easy it will be to do "go to definition" or "find all references" queries in this scheme. In general, all our tooling will have to be adapted to correctly handle the new format.

**Comment by xeno-by** [comment]
@ShaneDelmore Do we still need this functionality for our internal work?

**Comment by ShaneDelmore** [comment]
Yes

**Comment by olafurpg** [comment]
Does this need to be part of the SemanticDB spec or can it be handled on your side? I'm still not sure what how exactly this change would enable the following use-case

> It could be useful to support multiple versions of a library, as well as to identify the source of the symbol to assist with imports or build file changes.

Can this be added to semanticidx instead?
